traffic_ops/client/cacheconfig.go

```diff
@@ -0,0 +1,69 @@
+package client
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import (
+	"encoding/json"
+)
+
+type CacheConfigResponse struct {
+	Response CacheConfig `json:"response"`
+}
+
+type CacheConfig struct {
+	Domain           string                       `json:"domain"`
+	CDN              string                       `json:"cdn"`
+	DeliveryServices []CacheConfigDeliveryService `json:"delivery_services"`
+	Parents          []CacheConfigParent          `json:"parents"`
+	AllowIP          []string                     `json:"allow_ip"`
+}
+
+type CacheConfigDeliveryService struct {
+	Protocol          int      `json:"protocol"`
+	QueryStringIgnore int      `json:"query_string_ignore"`
+	XMLID             string   `json:"xml_id"`
+	Type              string   `json:"type"`
+	OriginFQDN        string   `json:"origin_fdqn"`
+	Regexes           []string `json:"regexes"`
+	DSCP              int      `json:"dscp"`
+}
+
+type CacheConfigParent struct {
+	Host   string `json:"host"`
+	Domain string `json:"domain"`
+	Port   uint   `json:"port"`
+}
+
+// CacheConfig gets the configuration data for a cache
+func (to *Session) CacheConfig(cache string) (*CacheConfig, error) {
+	url := "/api/1.3/configs/cache/" + cache
```

Comment:
what exactly is the "cache" parameter supposed to be? hostname? if so, could you rename that parameter so it is clear?

also, how about something more like this endpoint so it is consistent with the conventions we're trying establish that follow the api guidelines - https://cwiki.apache.org/confluence/display/TC/API+Guidelines

/api/1.3/caches/:hostname/configs

Reply:
It is consistent with `/api/1.2/cdns/title-vi/configs/monitoring`. This is a config endpoint, not a raw data endpoint.

Reply:
sorry, i don't see the consistency. this one is pulling configs (more specifically monitoring configs) for the title-vi cdn - /api/1.2/cdns/title-vi/configs/monitoring

you're trying to pull "configs" for a cache, right? so imo this would maintain consistency, no?

/api/1.3/caches/:cache/configs

Reply:
>what exactly is the "cache" parameter supposed to be? hostname?
What else would "cache" be? It's a string, I'm having trouble imagining what else someone would assume it to be. Idiomatic Go prefers short names like C, not verbose names like Java.

From https://golang.org/doc/effective_go.html#names
>  Another short example is once.Do; once.Do(setup) reads well and would not be improved by writing once.DoOrWaitUntilDone(setup). Long names don't automatically make things more readable. A helpful doc comment can often be more valuable than an extra long name. 

I've added a doc comment to more strictly conform to Go Language Idioms.

Reply:
`/caches` suggests it's returning raw data, e.g. from a table. The `configs` in `monitoring` should really be moved to the root as well.

Really, config logic should be separated from the API, in a completely separate library or executable, with the API having the proper endpoints to get only the necessary data. But that's outside the scope of this PR. I'm just trying to get an endpoint in, that we need to efficiently get the data needed to configure a cache. The current endpoints do not provide a way to get only the required data, without unnecessary slowness and bandwidth.

Reply:
you've lost me. looking at your code it looks like you are building a json response by querying the database. looks like any other api so why not follow the routes convention we've established?

Reply:
>why not follow the routes convention we've established?
It does.

Reply:
I think what Jeremy is saying here is that your route is: `/api/1.3/configs/cache/ + cache`
and the "standard" is `/api/1.3/caches/:cache/configs`.  I don't really care one way or another, but consistency is always good.